joss/paper.md

```diff
@@ -0,0 +1,166 @@
+    affiliation: 1
+  - name: Aaron Spring
+    orcid: 0000-0003-0216-2241
+    affiliation: 2
+affiliations:
+ - name: Department of Atmospheric and Oceanic Sciences and Institute of Arctic and Alpine Research, University of Colorado Boulder, Boulder, Colorado USA
+   index: 1
+ - name: International Max Planck Research School on Earth System Modelling and Max Planck Institute for Meteorology, Hamburg, Germany
+   index: 2
+date: 16 October 2020
+bibliography: paper.bib
+---
+
+<!-- 982 words total -->
+
+# Summary
+<!-- 71 words -->
+Predicting extreme events and variations in weather and climate provides crucial
+information for economic, social, and environmental decision-making. However,
+quantifying prediction skill for multi-dimensional geospatial model output is
+computationally expensive and a difficult coding challenge. The large datasets
+(order gigabytes to terabytes) require parallel and out-of-memory computing to be
+analyzed efficiently. Further, aligning the many forecast initializations with differing
+observational products is a straight-forward, but exhausting and error-prone exercise
+for researchers.
+
+<!-- 68 words -->
+To simplify and standardize forecast verification across scales from hourly weather to
+decadal climate forecasts, we built `climpred`: a community-driven python package for
+computationally efficient and methodologically consistent verification of ensemble
+prediction models. The code base is maintained through open-source development. It
+leverages `xarray` [@Hoyer:2017] to anticipate core prediction ensemble dimensions
+(ensemble `member`, `init`ialization date and `lead` time) and `dask` [@dask] to
+perform out-of-memory and parallelized computations on large datasets.
+
+ <!-- 54 words -->
+`climpred` aims to offer a comprehensive set of analysis tools for assessing the quality
+of dynamical forecasts relative to verification products (_e.g._, observations,
+reanalysis products, control simulations). The package includes a suite of deterministic
+and probabilistic verification metrics that are constantly expanded by the community and
+are generally organized in our companion package, `xskillscore`.
+
+<!-- 161 words -->
+# Statement of Need
+While other climate verification packages exist (_e.g._, `s2dverification`
+[@Manubens:2018] written in R and `MurCSS` [@Illing:2014] written with python-based
+`CDO`-bindings [@CDO]), `climpred` is unique for many reasons. (1) It spans broad
+temporal scales of prediction, supporting the weather, subseasonal-to-seasonal (S2S),
+and seasonal-to-decadal (S2D) communities. (2) `climpred` supports `dask` [@dask] and
+thus works across all computational scales, from personal laptops to supercomputers
+(HPC), and in the cloud. This leads to, for example, verification of a global 5°x5°
+resolution climate prediction in less than one second, compared to the 8 minutes
+required in `MurCSS`.  (3) `climpred` is highly modular and supports the research
```

Comment:
Do you think it's okay to put it this way? Don't want this to be offensive to the MurCSS folks for all the work they did, by singling them out.

Reply:
Ryan Abernathy suggests it's okay, since that's what pushes software forward. The way I wrote it in the medium article is this, which nods to their contributions and the bias in benchmarking. Maybe we can adapt that shortly here:

> Due to the use of vectorized numpy code and parallel, out-of-memory computing with dask , climpred verifies forecasts more quickly than other forecast verification packages. This enables a truly interactive analysis experience for users. For example, climpred can verify a 5° x 5° global forecast model in 0.5 seconds, relative to the 8 minutes required by MurCSS. This means that a user can quickly change verification metrics and rapidly assess their results. However, note that [benchmarking is inherently biased](https://matthewrocklin.com/blog/work/2017/03/09/biased-benchmarks) and MurCSS is quite valuable for their rigorous replication of decadal climate prediction metrics and their seamless integration with the MiKlip decadal prediction project at the Max Planck Institute for Meteorology.

Reply:
`offensive to the MurCSS folks for all the work` thats also my concern. but also Ryans comment is true. however, I would say something like 5 or a few seconds. the 8 minutes verify ACC, conditional and unconditional bias and CRPSS_ES over three leads. my CPRSS_ES examples over 10 leads took 10 seconds. so overall I think ~5 seconds is a fair comparison estimate from our side. I wouldnt go into details on this but just say `a few seconds`.